spec/features/category_index_spec.rb

```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+
+RSpec.feature 'Category Index Page', type: :feature, js: true do
+  include Devise::Test::IntegrationHelpers
+
+  let!(:user) { User.create(email: 'test@example.com', password: 'password') }
+  let!(:category1) { Category.create(name: 'Test1 Category', user: @user) }
+  let!(:category2) { Category.create(name: 'Test2 Category', user: @user) }
+
+  before do
+    user.categories << [category1, category2]
+    login_as(user, scope: :user)
+  end
+
+  before(:each) { visit categories_path }
+
+  scenario 'Visiting the index path' do
+    expect(page).to have_content('Categories')
+  end
+
+  scenario 'allows signing out' do
+    click_link('Sign Out')
+    expect(page).to have_current_path(root_path)
+  end
+end
```

Comment:
- [ ] You have a failing test here kindly fix this for an error-free codebase.

![image](https://github.com/VaheStepanyan100/budget-app/assets/81932478/8c52c5f3-0176-44f7-bef1-f928297b3e58)
#### Note that I ran the tests thrice, the first time there were 2 failing tests, and for the last 2 times only 1 failing test. If this is not the case on your end mark it as resolved and leave a note for the next reviewer